### PR TITLE
Preparing for workshopification

### PR DESCRIPTION
**Pull request by Anticept**
This PR code needs a careful audit as it had a hardcoded kick in it and was removed. There's potentially useful code in this though. We're readying it for putting on workshop.

Yes, I know, potential license conflict, but being that this isn't for profit anyways, the fallout from a license conflict is effectively nil.

This is not to be merged without a careful examination!

TODO:
- [ ] Remove version checking. Don't even comment it out, just purge it.
- [ ] Look over carefully for other potential abuse
- [ ] Add travis script to build addon
- [ ] Insert correct ignore scripts
- [ ] Fix Invisible door when using door tool - `console:Error creating 'wired_door' (Make sure the file is AddCSLuaFile'd and there aren't any errors!)`

### CODE REVIEW

**Comment by Anticept** [comment]
todo:

Invisible door when using door tool
console:Error creating 'wired_door' (Make sure the file is AddCSLuaFile'd and there aren't any errors!)

**Comment by thegrb93** [comment]
You mean a backdoored workshop addon was found and removed. They way you say it here sounds like the repo had the backdoor.

**Comment by marchkarcz** [comment]
Just checked autorun/server/servermessage.lua, where the backdoor was found, and it's not there. Creed just put it in his workshop addon.

**Comment by marchkarcz** [comment]
^ although there could be other backdoors laying around, I was just saying Creeds backdoor probably isn't here

**Comment by Anticept** [comment]
> 
> 
> You mean a backdoored workshop addon was found and removed. They way you say it here sounds like the repo had the backdoor.

Correct.

**Comment by thegrb93** [comment]
What hardcoded kick are you talking about?

**Comment by Anticept** [comment]
https://github.com/wiremod/wire-extras/pull/88/commits/9d5d37ae2b6abdedd4fa7de6fcb0eaf102209733#diff-42dee2b7a9f7b151ffd2feddc526d1ddL13

**Comment by thegrb93** [comment]
Oh so this is code from that workshop release except without that part.

**Comment by Divran** [comment]
List of things that need to be fixed in this PR

- potentially erase update checker (checks if you have the latest version installed - useless feature imo since workshop updates automatically and wire extras itself doesn't update often enough for this feature to be useful)
- commented out copy of update checker needs to be erased
- `gmod_wire_boyancy` entity has removed AddCSLua for its own clientside files, and added AddCSLua for its serverside file
- large chunks of the holo anim and pose library has been erased. probably breaking it
- useless whitespace added to `gmod_wire_field_device/init.lua`
- very inefficient code added to `gmod_wire_field_device/init.lua` (line 608 and below)
- `gmod_wire_hudindicator_2` has added AddCSLua for its serverside file
- `gmod_wire_hudindicator_2/util/table.lua` has a copy of PrintTable for no reason
- useless change in `gmod_wire_ramcard_defaultbase/init.lua`
- useless change in `gmod_wire_realmagnet/cl_init.lua`
- The changes in `gmod_wire_realmagnet/init.lua` appear to be useless as well, but may need to be checked more thoroughly.
- `gmod_wire_rfid_implanter/init.lua` removed duplicator support (??)
- someone else will need to check if the changes made to `gmod_wire_xyzbeacon/init.lua` don't break anything
- the changes made to `weapons/gmod_tool/stools/gui_wiring.lua` need to be undone entirely.
- useless changes in `gmod_tool/stools/multi_wire.lua` (useless whitespace and functions renamed to their alias)
- in several cases the changes in `gmod_tool/stools/multi_wire.lua` also try to use the beam library which no longer exists
- incorrect tool category in `gmod_tool/stools/wire_adv_hud_indicator_2.lua`
- useless whitespace changes in `gmod_tool/stools/wire_field_device.lua`
- potentially incorrect tool category in `gmod_tool/stools/wire_keycard.lua`. It might fit there better, I dunno.
- TargetOnlyMetal setting removed from `gmod_tool/stools/wire_magnet.lua` for some strange reason

TLDR: there's more things that need to be removed from this PR than things we can keep

**Comment by Divran** [comment]
my response to this PR https://i.kym-cdn.com/photos/images/newsfeed/001/354/591/17c.png

**Comment by Anticept** [comment]
We'll make our own then.
